app/src/main/java/ch/epfl/balelecbud/PicturesActivity.java

```diff
@@ -0,0 +1,130 @@
+package ch.epfl.balelecbud;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.ContentResolver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.View;
+import android.webkit.MimeTypeMap;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.core.content.FileProvider;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+public class PicturesActivity extends BasicActivity {
+
+    ImageView imageView;
+    Button cameraButton;
+    String currentPhotoPath;
+
+    public static final int CAMERA_PERM_CODE = 101;
+    public static final int CAMERA_REQUEST_CODE = 102;
+    public static final int GALLERY_REQUEST_CODE = 105;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_pictures);
+        configureToolBar(R.id.pictures_activity_toolbar);
+        configureDrawerLayout(R.id.pictures_activity_drawer_layout);
+        configureNavigationView(R.id.pictures_activity_nav_view);
+
+        cameraButton = findViewById(R.id.takePicBtn);
+        imageView = findViewById(R.id.picturesImageView);
+
+        cameraButton.setOnClickListener(v -> askCameraPermissions());
+
+    }
+
+    private void askCameraPermissions() {
+        if(ContextCompat.checkSelfPermission(this, Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED){
+            ActivityCompat.requestPermissions(this,new String[] {Manifest.permission.CAMERA}, CAMERA_PERM_CODE);
+        }else {
+            dispatchTakePictureIntent();
+        }
+
```

Comment:
If the user has already granted permissions to the app, then subsequent calls to `requestPermissions` will result in a call to `onRequestPermissionsResult` without prompting the user. So since you already call `dispatchTakePictureIntent` in `onRequestPermissionsResult` it will do the same by simply calling `requestPermissions`
```suggestion
        ActivityCompat.requestPermissions(this,new String[] {Manifest.permission.CAMERA}, CAMERA_PERM_CODE);
```